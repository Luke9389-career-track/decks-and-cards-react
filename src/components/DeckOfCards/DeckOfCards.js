import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import Deck from '../Deck/Deck';

function DeckOfCards({ title, items }) {

  return (
    <>
      <Deck title={title}>
        {items.map(({ _id, name, image }) => (
          <Card _id={_id} key={_id} title={name ? name : _id} image={image} content={name ? `${name} has ${_id}` : null}/>
        ))}
      </Deck>
    </>
  );
}

DeckOfCards.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string,
    name: PropTypes.string,
    image: PropTypes.string
  }))
};

export default DeckOfCards;




