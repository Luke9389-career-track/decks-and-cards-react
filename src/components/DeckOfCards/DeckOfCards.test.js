import React from 'react';
import { shallow } from 'enzyme';
import DeckOfCards from './DeckOfCards';

describe('DeckOfCards component', () => {
  it('renders DeckOfCards', () => {
    const wrapper = shallow(<DeckOfCards title='Test' items={[{ stuff: 'stuff' }, { stuff: 'stuff' }, { stuff: 'stuff' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
