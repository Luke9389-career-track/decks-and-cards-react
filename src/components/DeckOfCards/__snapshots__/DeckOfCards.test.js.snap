// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOfCards component renders DeckOfCards 1`] = `
<Fragment>
  <Deck
    title="Test"
  >
    <Card
      content={null}
    />
    <Card
      content={null}
    />
    <Card
      content={null}
    />
  </Deck>
</Fragment>
`;
